piksi_tools/console/settings_view.py

```diff
@@ -601,66 +644,72 @@ def _settings_import_from_file_button_fired(self):
+        parser = configparser.ConfigParser()
+        # the optionxform is needed to handle case sensitive settings
+        parser.optionxform = str
+        try:
+            with open(file.path, 'r') as f:
+                parser.read_file(f)
+        except configparser.ParsingError as e:  # file formatted incorrectly
+            print('Unable to parse ini file due to ParsingError: {}.'.format(e))
+            print('Unable to import settings to device.')
+            return
+        except IOError as e:  # IOError (likely a file permission issue)
+            print('Unable to read ini file due to IOError: {}'.format(e))
+            print('Unable to import settings to device.')
+            return
+
+        # Iterate over each setting
+        for section, settings in parser.items():
+
+            if not settings:
+                # Empty
+                continue
+
+            this_section = self.settings.get(section, None)
+
+            if this_section is None:
+                self._import_failure_section(section)
                 return
 
-            # Iterate over each setting and set in the GUI.
-            # Use the same mechanism as GUI to do settings write to device
-
-            for section, settings in parser.items():
-                this_section = self.settings.get(section, None)
-                for setting, value in settings.items():
-                    if this_section:
-                        this_setting = this_section.get(setting, None)
-                        if this_setting:
-                            this_setting.value = value
-                        else:
-                            print(("Unable to import settings from file. Setting \"{0}\" in section \"{1}\""
-                                   " has not been sent from device.").format(setting, section))
-                            return
-                    else:
-                        print(("Unable to import settings from file."
-                               " Setting section \"{0}\" has not been sent from device.").format(section))
-                        return
-
-            # Double check that no settings had a write failure.  All settings should exist if we get to this point.
-            a = TimedDelayStoppableThread(SETTINGS_REVERT_TIMEOUT + 0.1,
-                                          target=self._wait_for_any_write_failures,
-                                          args=(dict(parser)))
-            a.start()
-        else:
-            pass  # No error message here because user likely pressed cancel when choosing file
+            for setting, value in settings.items():
+                this_setting = this_section.get(setting, None)
+                if this_setting is None:
+                    self._import_failure_not_found(section, setting)
+                    return
+
+                if this_setting.value == value:
+                    continue
+
+                this_setting._write_value(this_setting.value, value)
```

Comment:
What happens if user imports a setting that breaks communication? How long will it take to timeout?

Reply:
With current defines each request is tried 5 times each having 500ms timeout so 2.5 seconds total.